src/app/services/error.service.ts

```diff
@@ -0,0 +1,49 @@
+import { ApolloError } from "@apollo/client/core";
+import { Injectable } from "@angular/core";
+import { ModalService } from "../components/modal/modal.service";
+import { logError } from "../common/app.helpers";
+import { ErrorTexts } from "../common/errors.text";
+import { NavigationService } from "./navigation.service";
+
+@Injectable({
+    providedIn: "root",
+})
+export class ErrorService {
+    private errorDescription = "";
+
+    constructor(
+        private modalService: ModalService,
+        private navigationService: NavigationService,
+    ) {}
+
+    public processError(error: Error, datasetName?: string): void {
+        const errorMapper: { [key in string]: string } = {
+            [ErrorTexts.ERROR_SERVER_BAD_SQL_QUERY]:
+                ErrorTexts.ERROR_SERVER_BAD_SQL_QUERY,
+            [ErrorTexts.ERROR_DATASET_NOT_FOUND]: `${datasetName ?? ""} ${
+                ErrorTexts.ERROR_DATASET_NOT_FOUND
+            }`,
+        };
```

Comment:
This string-to-string mapping seems very strange to me. Don't know what's considered a best practice in JS, but in other languages I would have an error hierarchy:
```
NotFoundError: Error {...}
DatasetNotFoundError: NotFoundError {...}

InvalidInputError: Error {...}
InvalidSQLError: InvalidInputError {...}
```
This way logic will only raise typed errors, e.g.:
```
new DatasetNotFoundError(datasetName)
```
and all message formatting will be left in this service:
```
if (error instanceof DatasetNotFoundError) {
  this.errorDescription = localize(ErrorMessages.ERROR_DATASET_NOT_FOUND, error.datasetName)
}
```

I think just the fact that your function signature is `processError(error: Error, datasetName?: string):` indicates that something is very wrong here. What would it look like in the long term?

```
processError(
  error: Error, datasetName?: string, 
  userName?: string, 
  organizationName?: string, 
  sqlQuery?: string, 
  blockHash?: string, 
  pageNumber?: string, 
  ...):
```
... you get the idea.

Reply:
Typed errors would make sense only if we plan implementing a Visitor pattern for error handling. While we don't, at least that does not look like a reasonable effort. 

But maybe a universal error structure like this:

```
class KamuError : Error {
    errorCode: string, // could be an enum, but TS enums are quire weird
    params: string[];
};
```

and then a dictionary with correspondence between possible values of `errorCode` and a reporting template string.